### ISSUE TITLE
Error running hot-restart: NoSuchMethodError: The getter 'assetPathsToEvict' was called on null

### ISSUE THREAD

**Issue by Ahmadre** [flutter/flutter#84945]
## Steps to Reproduce

1. Be on master channel (2.3.0-17.0.pre.411)
2. Run your app on iOS, Android or Web.
3. After app is fully started -> Hot-Restard/Hot-Reload your app.
4. App crashes!

**Expected results:** 
App is not crashing on hot-restart or hot-reload.

**Actual results:** 
App is crashing on hot-restart or hot-reload.

<details>
<summary>Error-Details</summary>

```bash
Lost connection to device.
Error running hot-restart: NoSuchMethodError: The getter 'assetPathsToEvict' was called on null.
Receiver: null
Tried calling: assetPathsToEvict
Exited (sigterm)
```

</details>

<details>
  <summary>Logs</summary>

```
[        ] {"DTCompiler":"com.apple.compilers.llvm.clang.1_0","CFBundleInfoDictionaryVersion":"6.0","UISupportedInterfaceOrientations~ipad":["UIInterfaceOrientationPortrait","UIInterfaceOrientationPortraitUpsideDown","UIInterfaceOrientationLandscapeLeft","UIInterfaceOrientationLandscapeRight"],"DTPlatformVersion":"14.5","CFBundleName":"Kack dich schlank","DTSDKName":"iphonesimulator14.5","UIViewControllerBasedStatusBarAppearance":false,"CFBundleIcons":{"CFBundlePrimaryIcon":{"CFBundleIconFiles":["AppIcon60x60"],"CFBundleIconName":"AppIcon"}},"LSRequiresIPhoneOS":true,"LSApplicationQueriesSchemes":["fbapi","fbapi20130214","fbapi20130410","fbapi20130702","fbapi20131010","fbapi20131219","fbapi20140410","fbapi20140116","fbapi20150313","fbapi20150629","fbapi20160328","fbauth","fbauth2"],"DTSDKBuild":"18E182","CFBundleShortVersionString":"0.0.23","CFBundleSupportedPlatforms":["iPhoneSimulator"],"UISupportedInterfaceOrientations":["UIInterfaceOrientationPortrait","UIInterfaceOrientationLandscapeLeft","UIInterfaceOrientationLandscapeRight"],"BuildMachineOSBuild":"20E241","DTPlatformBuild":"18E182","CFBundlePackageType":"APPL","DTXcodeBuild":"12E262","CFBundleDevelopmentRegion":"en","MinimumOSVersion":"11.0","CFBundleVersion":"16","UIStatusBarHidden":false,"FacebookAppID":"790358804929229","UIDeviceFamily":[1,2],"UILaunchStoryboardName":"prodLaunchScreen","CFBundleIdentifier":"de.kds.app","FacebookDisplayName":"Kack dich schlank","DTXcode":"1250","CFBundleExecutable":"Runner","CFBundleIcons~ipad":{"CFBundlePrimaryIcon":{"CFBundleIconFiles":["AppIcon60x60","AppIcon76x76"],"CFBundleIconName":"AppIcon"}},"CFBundleSignature":"????","UIMainStoryboardFile":"Main","DTPlatformName":"iphonesimulator","ITSAppUsesNonExemptEncryption":false,"CFBundleURLTypes":[{"CFBundleURLSchemes":["fb790358804929229"]},{"CFBundleTypeRole":"Editor","CFBundleURLSchemes":["com.googleusercontent.apps.358379537045-ua67b2pomi9gkl38r2ohhnara8qhlrjl"]},{"CFBundleTypeRole":"Editor","CFBundleURLSchemes":["com.googleusercontent.apps.358379537045-e6957ole2ska4hal4evipi08vr71s4kj"]}]}
[   +1 ms] executing: xcrun simctl launch 53C174C0-2F9C-4CC9-997C-BBD24CAB0EF5 de.kds.app --enable-dart-profiling --enable-checked-mode --verify-entry-points --start-paused --observatory-port=0
[ +243 ms] de.kds.app: 29073
[        ] Waiting for observatory port to be available...
[ +713 ms] Observatory URL on device: http://127.0.0.1:60325/tyveImORyzk=/
[  +10 ms] Caching compiled dill
[  +64 ms] Connecting to service protocol: http://127.0.0.1:60325/tyveImORyzk=/
[ +185 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:60325/tyveImORyzk=/.
[ +100 ms] DDS is listening at http://127.0.0.1:60332/1fv0_6ZnKZc=/.
[  +63 ms] Successfully connected to service protocol: http://127.0.0.1:60325/tyveImORyzk=/
[  +21 ms] DevFS: Creating new filesystem on the device (null)
[  +33 ms] DevFS: Created new filesystem on the device (file:///Users/rebar/Library/Developer/CoreSimulator/Devices/53C174C0-2F9C-4CC9-997C-BBD24CAB0EF5/data/Containers/Data/Application/0357B852-32E1-4924-840C-8BAC09018F8C/tmp/kack-dich-schlankQZNbnn/kack-dich-schlank/)
[   +3 ms] Updating assets
[ +162 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more information see #56466.
[   +7 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:app/main_prod.dart 57d0c858-1cac-4f7e-b326-bae43757c3b3
package:app/main_prod.dart:1
[        ] <- 57d0c858-1cac-4f7e-b326-bae43757c3b3
[ +122 ms] Updating files.
[        ] DevFS: Sync finished
[   +1 ms] Synced 0.0MB.
[   +1 ms] <- accept
[   +5 ms] Connected to _flutterView/0x7ff51f01f820.
Connecting to VM Service at ws://127.0.0.1:60332/1fv0_6ZnKZc=/ws

```

```
No issues found!
```

```
❯ flutter doctor -v
[✓] Flutter (Channel master, 2.3.0-17.0.pre.411, on macOS 11.3.1 20E241 darwin-x64, locale de-DE)
    • Flutter version 2.3.0-17.0.pre.411 at /Users/rebar/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 25ba467b5a (vor 2 Tagen), 2021-06-18 21:24:02 -0400
    • Engine revision ced58ef9c1
    • Dart version 2.14.0 (build 2.14.0-217.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 56.0.3
    • Dart plugin version 203.8173

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (4 available)
    • iPhone von Rebar (mobile)  • 00008030-001E6C520E07802E            • ios            • iOS 14.6
    • iPhone 12 Pro Max (mobile) • 53C174C0-2F9C-4CC9-997C-BBD24CAB0EF5 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 11.3.1
      20E241 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome
      91.0.4472.106
    ! Error: iPhone von Rebar is busy: Copying cache files from device. Xcode will continue when iPhone von
      Rebar is finished. (code -10)

! Doctor found issues in 1 category.
```

</details>

**Comment by darshankawar** [comment]
@Ahmadre 
Tried a sample app on latest master (2.3.0-17.0.pre.414) and hot-restarting / hot-reloading worked well without throwing the mentioned error:

```
dhs@Dhss-MacBook-Pro macbug % flutter run
Downloading Web SDK...                                              7.0s
Downloading darwin-x64/FlutterMacOS.framework tools...              4.3s
Downloading darwin-x64-profile/FlutterMacOS.framework tools...      1,109ms
Downloading darwin-x64-profile tools...                            629ms
Downloading darwin-x64-release/FlutterMacOS.framework tools...        982ms
Downloading darwin-x64-release tools...                            511ms
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running pod install...                                           2,113ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         7.6s
Xcode build done.                                           30.2s
Syncing files to device iPhone 12 Pro Max...                       103ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on iPhone 12 Pro Max is available at: http://127.0.0.1:54912/flvrm1MembU=/
The Flutter DevTools debugger and profiler on iPhone 12 Pro Max is available at: http://127.0.0.1:9104?uri=http%3A%2F%2F127.0.0.1%3A54912%2Fflvrm1MembU%3D%2F

Performing hot restart...                                               
Restarted application in 339ms.

Performing hot restart...                                               
Restarted application in 352ms.

Performing hot reload...                                                
Reloaded 0 libraries in 102ms.

Performing hot reload...                                                
Reloaded 0 libraries in 51ms.
 
```

Also verified the same on Web and both commands worked well.  Can you try to upgrade to latest master and see if issue still persists ?

<details>
<summary> flutter doctor -v </summary>

```
dhs@Dhss-MacBook-Pro ncflutter % flutter doctor -v
[✓] Flutter (Channel master, 2.3.0-17.0.pre.414, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.3.0-17.0.pre.414 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 757c3add14 (7 hours ago), 2021-06-20 22:29:02 -0400
    • Engine revision 9520bb15b3
    • Dart version 2.14.0 (build 2.14.0-228.0.dev)

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 91.0.4472.106

• No issues found!

```

</details>

**Comment by Ahmadre** [comment]
@darshankawar seems to be gone :) thx
